components/trusted-app-mgt/org.wso2.carbon.identity.trusted.app.mgt/src/main/java/org/wso2/carbon/identity/trusted/app/mgt/servlet/IosTrustedAppDiscoveryServlet.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.trusted.app.mgt.servlet;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.trusted.app.mgt.exceptions.TrustedAppMgtException;
+import org.wso2.carbon.identity.trusted.app.mgt.internal.TrustedAppMgtDataHolder;
+import org.wso2.carbon.identity.trusted.app.mgt.model.TrustedIosApp;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.APPS_ATTRIBUTE;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.CT_APPLICATION_JSON;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.HTTP_RESP_HEADER_CACHE_CONTROL;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.HTTP_RESP_HEADER_PRAGMA;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.HTTP_RESP_HEADER_VAL_CACHE_CONTROL_NO_STORE;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.HTTP_RESP_HEADER_VAL_PRAGMA_NO_CACHE;
+import static org.wso2.carbon.identity.trusted.app.mgt.utils.Constants.IOS_CREDENTIAL_PERMISSION;
+
+/**
+ * Servlet to discover iOS based trusted mpbile apps.
+ */
+public class IosTrustedAppDiscoveryServlet extends HttpServlet {
```

Comment:
Lets add a unit test to this class. We can have a predefined json response and then validate if the response coming from here is the same.